panel_app/panel_UI/step2_downscale.py

```diff
@@ -152,15 +204,43 @@ def step2_region_view():
     # Force create a fresh map widget to avoid reference conflicts
     map_widget = get_map_widget(force_new=True)
     controls = get_controls()
+    obs_toggle = controls.get("obs_domain", None)
+
+    if obs_toggle is not None:
+
+        def _on_obs_domain_change(event):
```

Comment:
How do we feel about nested functions in python? These two seem pretty big to be nested inside another one...